lib/take_away.rb

```diff
@@ -0,0 +1,68 @@
+class TakeAway
+  
+  attr_reader :menu, :order, :take_away
+
+  def initialize(menu = Menu.new, confirmation = Confirmation.new)
+    @confirmation = confirmation
+    @menu = menu
+    @order = []
+    @take_away = 0
+    @total = 0
+    @current_dish = []
+  end
+
+  def view_menu
+    @menu.view
+  end
+
+  def add_dish(number, qty)
+    fail "please enter a valid dish number" if @menu.existing_dish(number, qty) != true
+    if duplicate_dish(number, qty) == true
```

Comment:
This seems like a good idea. I just kept adding to the 'basket' array of hashes, so your idea of actually editing a quantity value seems a lot more like a real user experience